pyfar/classes/audio.py

```diff
@@ -232,18 +239,31 @@ def time(self, value):
         """Set the time data."""
         # check and set the data and meta data
         data = np.atleast_2d(np.asarray(value))
-        if data.dtype.kind == "i":
-            data = data.astype("float")
-        elif data.dtype.kind != "f":
-            raise ValueError(
-                f"time data is {data.dtype}  must be int or float")
-        data = np.atleast_2d(np.asarray(value, dtype=float))
+        if self.complex:
+            data = data.astype("complex")
+            data = np.atleast_2d(np.asarray(value, dtype=complex))
```

Comment:
```suggestion
```
looks like this is redundant

Reply:
Thanks for the review. You're right here. I removed that line.